assets/js/base/utils/legacy-events.js

```diff
@@ -3,30 +3,40 @@ const Event = window.Event || null;
 /**
  * Wrapper function to dispatch an event so it's compatible with IE11.
  *
- * @param {string}  name       Name of the event to dispatch.
- * @param {boolean} bubbles    Whether the event bubbles.
- * @param {boolean} cancelable Whether the event is cancelable.
+ * @param {string}    name                 Name of the event to dispatch.
+ * @param {Object}    [options]            Some additional options to modify
+ *                                         the event.
+ * @param {boolean}   [options.bubbles]    Whether the event bubbles.
+ * @param {boolean}   [options.cancelable] Whether the event is cancelable.
+ * @param {HTMLNode}  [options.element]    Element that dispatches the event. By
+ *                                         default, the body.
  */
-const dispatchEvent = ( name, bubbles = false, cancelable = false ) => {
+export const dispatchEvent = (
+	name,
+	{ bubbles = false, cancelable = false, element }
+) => {
+	if ( ! element ) {
+		element = document.body;
```

Comment:
why not have this in the destructuring declaration?

Reply:
Ah, I don't know why. I thought it was breaking e2e tests but now it seems to work fine. Changed it in 04ae943c675b94e599a1d1700075a79ef8be5004.

Reply:
Had to revert this. Taking a closer inspection what happened was:
* `document.body.querySelector( ... )` returned `null` if it didn't find the element, so `element` was null.
* when destructuring a nullish object property, the default value doesn't seem to be applied, so `element` was not getting assigned to `document.body`, instead it kept the `null` value.

TIL
```JS
const wizard = { name: 'Harry', middleName: null, lastName: 'Potter'};
const { middleName = 'James' } = wizard;
console.log( middleName ); // null
```

Reply:
👍 if `middleName` was `undefined` however, then `James` would be the value. `null` is considered a value which is why the default doesn't get assigned.